File: C++/Bicchieri_Oggetti/Bicchiere.cpp
#include "Bicchiere.h"
#include <string>

using namespace std;

Bicchiere::Bicchiere() {

}

Bicchiere::Bicchiere(string iMateriale, string iForma, float iLivello, float iCapacita) {
    materiale = iMateriale;
    forma = iForma;
    livello = iLivello;
    capacita = iCapacita;
}

void Bicchiere::setMateriale(string iMateriale) {
    materiale = iMateriale;
}

void Bicchiere::setForma(string iForma) {
    forma = iForma;
}

void Bicchiere::setLivello(float iLivello) {
    livello = iLivello;
}

void Bicchiere::setCapacita(float iCapacita) {
    capacita = iCapacita;
}

string Bicchiere::getMateriale() {
    return materiale;
}

string Bicchiere::getForma() {
    return forma;
}

float Bicchiere::getLivello() {
    return livello;
}

float Bicchiere::getCapacita() {
    return capacita;
}

void Bicchiere::svuota() {
    livello = 0;
}

void Bicchiere::riempi() {
    livello = capacita;
}

File: Java/Bicchiere_Simo/src/me/simo/Bicchiere.java
package me.simo;

public class Bicchiere
{
    private String forma;
    private String materiale;
    private float capacita;
    private float livello;

    public Bicchiere(){}
    public Bicchiere(String forma, String materiale, float capacita, float livello){}
    //public ~Bicchiere(){}
    public void setLivello(float livello) {
        this.livello = livello;
    }
    public void setForma(String forma) {
        this.forma = forma;
    }
    public void setCapacita(float capacita) {
        this.capacita = capacita;
    }
    public void setMateriale(String materiale) {
        this.materiale = materiale;
    }
    public float getCapacita() {
        return capacita;
    }
    public float getLivello() {
        return livello;
    }
    public String getForma() {
        return forma;
    }
    public String getMateriale() {
        return materiale;
    }
    public void riempi(){
        livello=capacita;
    }
    public void svuota(){
        livello=0;
    }
}
